client.go

```diff
@@ -212,8 +232,16 @@ func (d *Dialer) Dial(urlStr string, requestHeader http.Header) (*Conn, *http.Re
 	// Get network dial function.
```

Comment:
Set `deadline` to minimum of ctx.Deadline() and time computed from d.HandshakeTimeout.